### PR TITLE
Serialization improvements

### PR DESCRIPTION
**Pull request by samgreen**

### CODE REVIEW

File: Mixpanel/Mixpanel.m [hunk 1]
```diff
@@ -864,8 +864,7 @@ - (id)unarchiveFromFile:(NSString *)filePath
     @catch (NSException *exception) {
         MixpanelError(@"%@ unable to unarchive data in %@, starting fresh", self, filePath);
         unarchivedData = nil;
-    }
-    if ([[NSFileManager defaultManager] fileExistsAtPath:filePath]) {
+
```
Comment (yarneo) [inline Mixpanel/Mixpanel.m]: remove space

File: Mixpanel/Mixpanel.m [hunk 2]
```diff
@@ -749,8 +755,19 @@ - (NSTimeInterval)retryBackOffTimeWithConsecutiveFailures:(NSUInteger)failureCou
 }
 
 #pragma mark - Persistence
+- (uint64_t)remainingFreeSpaceInBytes
```
Comment (patedit) [inline Mixpanel/Mixpanel.m]: We don't need this for now :)

File: Mixpanel/Mixpanel.m [hunk 3]
```diff
@@ -563,14 +562,15 @@ - (void)reset
...
-        self.peopleQueue = [NSMutableArray array];
         self.timedEvents = [NSMutableDictionary dictionary];
         self.shownSurveyCollections = [NSMutableSet set];
         self.shownNotifications = [NSMutableSet set];
         self.decideResponseCached = NO;
         self.variants = [NSSet set];
         self.eventBindings = [NSSet set];
+        
+        [self archiveAllEvents:[NSMutableArray array]];
+        [self archiveAllPeople:[NSMutableArray array]];
```
Comment (patedit) [inline Mixpanel/Mixpanel.m]: Seems that every time we call `[self archive]` we always call `[self archiveAllEvents:]` and `[self archiveAllPeople:]`. Looks reasonable to me to include `archiveAllEvents` and `archiveAllPeople` inside `archive` and we pass the `NSMutableArray` as params. What do you think? If we do that, we can modify this line and `flushWithCompletionHandler` method as well.

File: Mixpanel/Mixpanel.m [hunk 4]
```diff
@@ -863,32 +890,34 @@ - (id)unarchiveFromFile:(NSString *)filePath
...
-            MixpanelError(@"%@ unable to remove archived file at %@ - %@", self, filePath, error);
+        if (!removed || error) {
+            MixpanelError(@"%@ unable to remove archived file at %@ - Details: %@", self, filePath, error);
         }
     }
     return unarchivedData;
 }
 
-- (void)unarchiveEvents
+- (NSMutableArray *)unarchiveEvents
```
Comment (patedit) [inline Mixpanel/Mixpanel.m]: For consistency, `unarchiveAllEvents`? Same with `people`.

File: Mixpanel/Mixpanel.m [hunk 5]
```diff
@@ -651,14 +651,20 @@ - (void)flushWithCompletion:(void (^)())handler
 
 - (void)flushEvents
 {
-    [self flushQueue:_eventsQueue
-            endpoint:@"/track/"];
+    NSMutableArray *events = [self unarchiveEvents];
+    [self flushQueue:events
+            endpoint:kEndpointTrack];
+    
+    [self archiveAllEvents:events];
```
Comment (patedit) [inline Mixpanel/Mixpanel.m]: Following my first comment, this is a good candidate to change as well. `flushEvents` will just flush and can return the events that are not flushed. Those events will be captured on `flushWithCompletion` that will just call `archive`.

**Pull request merged**